app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManagerHolder.kt

```diff
@@ -1,17 +1,22 @@
 package org.p2p.wallet.swap.jupiter.statemanager
 
+import timber.log.Timber
+
 class SwapStateManagerHolder {
 
     private val swapStateManagerHolder = mutableMapOf<String, SwapStateManager>()
 
-    fun getOrCreate(key: String, defaultValue: () -> SwapStateManager): SwapStateManager =
-        swapStateManagerHolder.getOrPut(key) {
+    fun getOrCreate(key: String, defaultValue: () -> SwapStateManager): SwapStateManager {
+        return swapStateManagerHolder.getOrPut(key) {
+            Timber.i("Swap state manager not found for key $key; creating new")
```

Comment:
weird stuff here,
Why do you have multiple `SwapStateManager`'s?

Reply:
for each screens opened from tabs or as a fullscreen from token screen, for example